posthog/models/kafka_partition_stats/sql.py

```diff
@@ -83,14 +94,69 @@
 """
 )
 
-CREATE_KAFKA_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS = CREATE_PARTITION_STATISTICS_KAFKA_TABLE(
-    KAFKA_EVENTS_PLUGIN_INGESTION
-)
-
 CREATE_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS_MV = CREATE_PARTITION_STATISTICS_MV(KAFKA_EVENTS_PLUGIN_INGESTION)
 
-DROP_KAFKA_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS = DROP_PARTITION_STATISTICS_KAFKA_TABLE(
-    KAFKA_EVENTS_PLUGIN_INGESTION
-)
-
 DROP_EVENTS_PLUGIN_INGESTION_PARTITION_STATISTICS_MV = DROP_PARTITION_STATISTICS_MV(KAFKA_EVENTS_PLUGIN_INGESTION)
+
+# V2
+
+
+class PartitionStatsV2Table:
+    table_name: str = "events_plugin_ingestion_partition_statistics_v2"
+
+    def get_create_table_sql(self) -> str:
+        engine = ReplacingMergeTree(self.table_name, ver="timestamp")
+        return f"""
+            CREATE TABLE IF NOT EXISTS `{CLICKHOUSE_DATABASE}`.{self.table_name} ON CLUSTER '{CLICKHOUSE_CLUSTER}' (
+                topic LowCardinality(String),
+                partition UInt64,
+                offset UInt64,
```

Comment:
Delta (for partition) and double delta (for offset) encoding here would probably yield pretty significant compression here - can either try those here, or add them later to get a more interesting before/after comparison.

Reply:
I wouldn't worry too much about size on these tables since we are TTLing after 30 days.
I'm pretty sure if you were to add delta or double delta here it would just be duplicating data since you have `offset` in your sort key here and ClickHouse would definitely not put a delta or double delta encoded output in the sort key on disk.